server/types/internal_char.go

```diff
@@ -131,12 +131,8 @@ func (b InternalCharType) GetSerializationID() SerializationID {
 
 // IoInput implements the DoltgresType interface.
 func (b InternalCharType) IoInput(ctx *sql.Context, input string) (any, error) {
-	input, runeLength := truncateString(input, InternalCharLength)
-	if runeLength < InternalCharLength {
-		return input + strings.Repeat(" ", int(InternalCharLength-runeLength)), nil
-	} else {
-		return input, nil
-	}
+	input, _ = truncateString(input, InternalCharLength)
+	return input, nil
```

Comment:
Since `InternalCharLength` is only `1`, we can change this to just truncate to the first character (unicode character, rather than first byte), or return `" "` if the input is a zero-length string.

Reply:
Since we're using `unknown` type for empty string literel `''` converted to `char` type using `IoOutput` and `IoInput` functions, the empty string literal gets result of `' '`, which is caught by this test: https://github.com/dolthub/doltgresql/blob/main/testing/go/types_test.go#L289

Reply:
I see, so internal char does not pad like bpchar does, which is a mistake of mine. In that case, we can simplify it so that it trims to 1 unicode character if it's larger than that. We can add a test too to ensure that it trims to unicode rather than byte:
`INSERT INTO t_char VALUES (6, 'こんにちは')`